node/src/components/contract_runtime/operations.rs

```diff
@@ -48,47 +162,33 @@ pub(super) async fn commit_execution_effects(
             effect
         }
     };
-    let commit_result = commit(
+    let new_state_root = commit(
         engine_state,
         metrics,
         state_root_hash,
         execution_effect.transforms,
-    )
-    .await;
-    trace!(?commit_result, "commit result");
-    match commit_result {
-        Ok(CommitResult::Success { state_root }) => {
-            debug!(?state_root, "commit succeeded");
-            Ok((state_root.into(), execution_result))
-        }
-        _ => {
-            error!(
-                ?commit_result,
-                "commit failed - internal contract runtime error"
-            );
-            Err(())
-        }
-    }
+    )?;
+    Ok((new_state_root, execution_result))
 }
 
-pub(super) async fn commit(
-    engine_state: Arc<EngineState<LmdbGlobalState>>,
-    metrics: Arc<ContractRuntimeMetrics>,
+fn commit(
```

Comment:
```suggestion
fn commit_transforms(
```

Reply:
Fixed in ab987022cd28168aa6451e947e27c3914366b66c